File: Phase 2/backend/api/model/problem.go
package model

import (
	"M-AI/api/constants"
	"gorm.io/gorm"
)

type Problem struct {
	gorm.Model
	Topic    constants.TopicEnum `gorm:"type:topic_enum" json:"topic"`
	Title    string              `json:"title"`
	Question string              `json:"question"`
}

File: Phase 2/frontend/lib/api/problem.ts
import axios from './axios';

export interface Problem {
    id: number;
    topic: string;
    title: string;
    question: string;
    createdAt: string;
    updatedAt: string;
}

export interface CreateProblemRequest {
    question: string;
}

const ProblemAPI = {
    createProblem: (data: CreateProblemRequest) => {
        return axios.post<Problem>('/problems', data, { withCredentials: true });
    },
};

export default ProblemAPI;